VimeoNetworking/Sources/VimeoRequestSerializer.swift

```diff
@@ -125,32 +125,32 @@ final public class VimeoRequestSerializer: AFJSONRequestSerializer
     
     // MARK: Private API
     
-    private func setup(apiVersion apiVersion: String)
+    private func setup(apiVersion: String)
     {
-        self.setValue("application/vnd.vimeo.*+json; version=\(apiVersion)", forHTTPHeaderField: self.dynamicType.AcceptHeaderKey)
+        self.setValue("application/vnd.vimeo.*+json; version=\(apiVersion)", forHTTPHeaderField: type(of: self).AcceptHeaderKey)
 //        self.writingOptions = .PrettyPrinted
     }
 
-    private func setAuthorizationHeader(request request: NSMutableURLRequest) -> NSMutableURLRequest
+    private func setAuthorizationHeader(request: NSMutableURLRequest) -> NSMutableURLRequest
```

Comment:
I'm not sure what to do here, @ghking. We're overriding a method from **AFURLRequestSerialization** which returns an **NSMutableURLRequest**.

```Swift
override public func request(withMethod method: String, urlString URLString: String, parameters: Any?, error: NSErrorPointer) -> NSMutableURLRequest
```

Inside we're calling through to a method that I've updated to accept a **URLRequest** and return a **URLRequest**. But now I have a situation where I start with an **NSMutableURLRequest**, but I've casted it as a URLRequest. But now I need to cast it _back_ to an **NSMutableURLRequest**. Which seems terrible and weird. Also, the cast can't be allowed to fail because the method we're overriding does not return an optional.

So where I'm at now is something like this:

```Swift
override public func request(withMethod method: String, urlString URLString: String, parameters: Any?, error: NSErrorPointer) -> NSMutableURLRequest
{
    let request = super.request(withMethod: method, urlString: URLString, parameters: parameters, error: error)
    
    let requestWithAuthHeader = self.setAuthorizationHeader(request: request as URLRequest)
    
    // Cast back to the original type.
    let mutableRequest = requestWithAuthHeader as! NSMutableURLRequest
   
    return request
}
```

I'm not sure if I should just keep the **NSMutableURLRequests** for now and expect we'll replace AFNetworking with AlamoFire, or do some weird casting dance instead. Any thoughts?